python/tank/descriptor/io_descriptor/appstore.py

```diff
@@ -107,15 +126,14 @@ def __init__(self, descriptor_dict, sg_connection, bundle_type):
         self._validate_descriptor(
             descriptor_dict,
             required=["type", "name", "version"],
-            optional=[]
+            optional=["label"]
```

Comment:
nope, version is required, but there are some cases (e.g. bootstrap) where you can provide descriptors without version numbers and the first thing that happens is that the latest version is resolved.